### DISCUSSION TITLE
LSM open loop cdf generation

### DISCUSSION THREAD

**Discussion by grthipavani** [NASA-LIS/LISF#840] [Q&A] [answered]
Hi,

 I did step 4 of DA preprocessing for generating the LSM soil moisture CDF files. The run was successful and the files got created. When I visualize using grads, soil moisture is having 0 values instead of cumulative distribution plot. I checked LIS_HIST file and soil moisture is in m3/m3 units. Kindly let me know the reason behind the above issue. I attached below the ldtlog, ldt.config below. 

Thank you,
Pavani
[file.zip](https://github.com/NASA-LIS/LISF/files/6832429/file.zip)

**Comment by dmocko** [comment, marked as answer]
Hi Pavani,

Can you please attach your cdf_noah36.nc file?

It's unclear to me from your question if the values in the netCDF file are zero, or just if the values are shown as zero when plotted in GrADS.

Did you adjust the values in the GrADS .xdf file and the .gs script to reflect your different domain than in the Step 4 end-to-end testcase?

I also noticed that you are using a monthly temporal resolution for the CDFs (the original testcase uses annual).

David

**Reply by grthipavani** [reply]
Hi,
I am unable to attach the cdf file as it exceeds the maximum size limit. 
Yes, I adjusted the values in the xdf domain that reflects the domain.
Since I used monthly temporal resolution, I tried plotting cdf vs  soil moisture for each month. **Expect for march month, all other months are having zero values**. I attached below the cdf plot for march month. 
Does the monthly cdf gives values for all the months ?

Thank you,
Pavani
![marchcdf](https://user-images.githubusercontent.com/87541491/126066299-22f3d367-ee55-405d-93b4-0a970abb8eff.png)

**Reply by dmocko** [reply]
Your config entry:

Observation count threshold:          30

Since you have daily "observations" of your modeled soil moisture, I can see how March (with 31 days) gives a CDF, while months with 30 days would not.

Can you try lowering this threshold?  The value of 30 is appropriate for yearly CDFs, not monthly.

David

**Reply by grthipavani** [reply]
Hi David,

I changed the **Observation count threshold** to 15 and I am able to get the CDF plot for all the months. 
Thanks for your help.
Is there any optimum value for Observation count threshold ?

Thank you,
Pavani

**Reply by dmocko** [reply]
Hi Pavani,

Sorry - I don't have any expertise on the optimum value.  Are you going to run over more than one year?

Any thoughts - @sujayvkumar @LIS-navari or @yhkwon81 on an optimum value for the threshold?

David

**Reply by yhkwon81** [reply]
Hi Pavani,

In my knowledge, there is no specific optimum value for the observation count threshold. More data points may lead to better CDFs. Thus, the use of higher threshold values may produce better CDFs, but more grid cells whose number of data points is less than the threshold may not have CDFs. The decision of the threshold should be based on the available time period of the data and revisit time of the satellite. 

Thanks,
Yonghwan

**Reply by grthipavani** [reply]
Hi David and yhkwon81,

Thank you. It is very helpful.
I am going to run for 8 months.

Thanks,
Pavani
